application.cpp

```diff
@@ -440,19 +492,40 @@ void application::set_thread_priority_max() {
 }
 
 void application::exec() {
+   std::exception_ptr eptr = nullptr;
    {
-      boost::asio::io_service::work work(*io_serv);
+      boost::asio::io_service::work work(io_serv);
       (void)work;
       bool more = true;
-      while( more || io_serv->run_one() ) {
-         while( io_serv->poll_one() ) {}
-         // execute the highest priority item
-         more = pri_queue.execute_highest();
+      
+      while( more || io_serv.run_one() ) {
+         if (my->_is_quiting)
+            break;
+         try {
+            while( io_serv.poll_one() ) {}
+            // execute the highest priority item
+            more = pri_queue.execute_highest();
+         } catch(...) {
+            more = false;
```

Comment:
Should `more` be set to the actual state of pri_queue by checking its size?

Reply:
If `more` is set at line 507, it is already set with the state of the `priority_queue` (whether it is empty or not, which is all that matters)
If `more` is set at line 509, we want it to be false, even if there are some items left in the priority_queue, because we want to exit immediately. This is a change of behavior that was discussed with Kevin and which is desired.

Reply:
I saw the discussions above. Was thinking why `more` needed to be set on line 509. As you have already called `quit()`, the loop will terminate because of _is_quiting.

Reply:
Thank you very much for insisting!
I had a better look at it, and you are right, the setting of `more` here is incorrect. We should set `more = true` so that we don't execute `io_serv.run_one()` on the `while` loop, and we will exit the loop immediately because of _is_quiting.